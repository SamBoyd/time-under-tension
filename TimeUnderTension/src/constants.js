import 'react-native-get-random-values';
import {v4 as uuidv4} from 'uuid';

export const PAGE = {
    main: 'main',
    workout: 'workout',
    pickExercise: 'pickExercise',
    createTemplateWorkout: 'createTemplateWorkout',
    manageExercises: 'manageExercises',
    addExercise: 'addExercise'
}

export const TIMER_STATE = {
    ready: 'ready',
    setup: 'setup',
    work: 'work',
    rest: 'rest'
}

export const DEFAULT_SOUND_FILE = 'buzzer.mp3'

export const DEFAULT_SETUP_TIME = 2
export const DEFAULT_WORK_TIME_LOWER = 5
export const DEFAULT_WORK_TIME_UPPER = 15
export const DEFAULT_REST_TIME = 3


export const exerciseCategory = {
    chest: 'chest',
    arms: 'arms',
    shoulders: 'shoulders',
    back: 'back',
    legs: 'legs'
}

export const exercises = [
    {
        id: uuidv4(),
        name: 'Bench press',
        category: exerciseCategory.chest
    },
    {
        id: uuidv4(),
        name: 'Dumbbell Fly',
        category: exerciseCategory.chest
    },
    {
        id: uuidv4(),
        name: 'Incline bench press',
        category: exerciseCategory.chest
    },
    {
        id: uuidv4(),
        name: 'Machine press',
        category: exerciseCategory.chest
    },
    {
        id: uuidv4(),
        name: 'Cable bench press',
        category: exerciseCategory.chest
    },

    {
        id: uuidv4(),
        name: 'Leg press',
        category: exerciseCategory.legs
    },
    {
        id: uuidv4(),
        name: 'Back squat',
        category: exerciseCategory.legs
    },
    {
        id: uuidv4(),
        name: 'Front squat',
        category: exerciseCategory.legs
    },
    {
        id: uuidv4(),
        name: 'Leg curl',
        category: exerciseCategory.legs
    },
    {
        id: uuidv4(),
        name: 'Calf raise',
        category: exerciseCategory.legs
    },

    {
        id: uuidv4(),
        name: 'Military press',
        category: exerciseCategory.shoulders
    },
    {
        id: uuidv4(),
        name: 'Lateral raise',
        category: exerciseCategory.shoulders
    },
    {
        id: uuidv4(),
        name: 'Arnold press',
        category: exerciseCategory.shoulders
    },

    {
        id: uuidv4(),
        name: 'Barbell curl',
        category: exerciseCategory.arms
    },
    {
        id: uuidv4(),
        name: 'Tricep extension',
        category: exerciseCategory.arms
    },
    {
        id: uuidv4(),
        name: 'Dips',
        category: exerciseCategory.arms
    },

    {
        id: uuidv4(),
        name: 'Lat pulldown',
        category: exerciseCategory.back
    },
    {
        id: uuidv4(),
        name: 'Row',
        category: exerciseCategory.back
    },
    {
        id: uuidv4(),
        name: 'Pull up',
        category: exerciseCategory.back
    },
]
