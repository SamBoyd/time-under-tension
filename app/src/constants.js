import {v4 as uuidv4} from 'uuid';

export const PAGE = {
    workout: 'workout',
    pickExercise: 'pickExercise'
}


export const exerciseCategory = {
    chest: 'chest',
    arms: 'arms',
    shoulders: 'shoulders',
    back: 'back',
    legs: 'legs'
}

export const exercises = [
    {
        id: uuidv4(),
        name: 'Bench press',
        category: exerciseCategory.chest
    },
    {
        id: uuidv4(),
        name: 'Dumbbell Fly',
        category: exerciseCategory.chest
    },
    {
        id: uuidv4(),
        name: 'Incline bench press',
        category: exerciseCategory.chest
    },
    {
        id: uuidv4(),
        name: 'Machine press',
        category: exerciseCategory.chest
    },
    {
        id: uuidv4(),
        name: 'Cable bench press',
        category: exerciseCategory.chest
    },

    {
        id: uuidv4(),
        name: 'Leg press',
        category: exerciseCategory.legs
    },
    {
        id: uuidv4(),
        name: 'Back squat',
        category: exerciseCategory.legs
    },
    {
        id: uuidv4(),
        name: 'Front squat',
        category: exerciseCategory.legs
    },
    {
        id: uuidv4(),
        name: 'Leg curl',
        category: exerciseCategory.legs
    },
    {
        id: uuidv4(),
        name: 'Calf raise',
        category: exerciseCategory.legs
    },

    {
        id: uuidv4(),
        name: 'Military press',
        category: exerciseCategory.shoulders
    },
    {
        id: uuidv4(),
        name: 'Lateral raise',
        category: exerciseCategory.shoulders
    },
    {
        id: uuidv4(),
        name: 'Arnold press',
        category: exerciseCategory.shoulders
    },

    {
        id: uuidv4(),
        name: 'Barbell curl',
        category: exerciseCategory.arms
    },
    {
        id: uuidv4(),
        name: 'Tricep extension',
        category: exerciseCategory.arms
    },
    {
        id: uuidv4(),
        name: 'Dips',
        category: exerciseCategory.arms
    },

    {
        id: uuidv4(),
        name: 'Lat pulldown',
        category: exerciseCategory.back
    },
    {
        id: uuidv4(),
        name: 'Row',
        category: exerciseCategory.back
    },
    {
        id: uuidv4(),
        name: 'Pull up',
        category: exerciseCategory.back
    },
]
