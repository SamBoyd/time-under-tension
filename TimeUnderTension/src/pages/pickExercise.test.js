import React from 'react';
import configureStore from 'redux-mock-store';
import {Provider} from "react-redux";
import renderer from "react-test-renderer";

import PickExercise from "./pickExercise";
import {PAGE} from "../constants";

jest.useFakeTimers()

const mockStore = configureStore([]);

describe('Pick Exercise for workout', () => {
    let store;
    let component;
    let navigation;
    let route;


    beforeEach(() => {
        store = mockStore({
            exercises: {
                exercises: [
                    {
                        "name": "Exercise 1",
                        "category": "cat_1"
                    },
                    {
                        "name": "Exercise 2",
                        "category": "cat_2"
                    }
                ],
                categories: ['cat_1', 'cat_2'],
                newExercise: {
                    'name': '',
                    'category': ''
                }
            }

        });

        navigation = jest.fn();
        navigation.navigate = jest.fn()
        route = {
            params: {
                saveWorkTo: 'workout'
            }
        }

        component = renderer.create(
            <Provider store={store}>
                <PickExercise
                    navigation={navigation}
                    route={route}
                />
            </Provider>
        );
    });

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('can move to creating a new exercise', () => {
        renderer.act(() => {
            component.root.findByProps({testId: 'newExerciseButton'}).props.onPress()
        })

        // expect(store.dispatch).toHaveBeenCalledTimes(0);

        expect(navigation.navigate).toHaveBeenCalledTimes(1)
        expect(navigation.navigate).toHaveBeenCalledWith(PAGE.addExercise)
    })
})