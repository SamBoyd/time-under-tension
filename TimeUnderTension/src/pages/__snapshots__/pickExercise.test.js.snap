// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pick Exercise for workout should render with given state from Redux store 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "backgroundColor": "#262521",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#262521",
        "flex": 1,
        "height": 1334,
        "paddingLeft": 37.5,
        "paddingRight": 37.5,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "paddingBottom": 200,
        }
      }
      overScrollMode="never"
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "textAlignVertical": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                Array [
                  undefined,
                  Object {
                    "height": 53.5,
                    "padding": 0,
                  },
                ],
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#31302C",
                    "borderColor": "#5F758E",
                    "borderRadius": 8,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 53.5,
                    "justifyContent": "center",
                    "padding": 8,
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#CECABA",
                      "fontSize": 12,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  back
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                Array [
                  undefined,
                  Object {
                    "height": 53.5,
                    "padding": 0,
                  },
                ],
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#31302C",
                    "borderColor": "#5F758E",
                    "borderRadius": 8,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 53.5,
                    "justifyContent": "center",
                    "padding": 8,
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#CECABA",
                      "fontSize": 12,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  New
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#262521",
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
                "paddingLeft": 75,
                "paddingTop": 40,
                "text": Object {
                  "color": "#CECABA",
                },
              }
            }
            testID="padView"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#CECABA",
                    "fontSize": 17,
                  }
                }
                testID="listItemTitle"
              >
                Cat_1
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingLeft": 16,
                }
              }
            />
            <View
              collapsable={false}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 24,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      󰅀
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Object {
              "maxHeight": "0%",
              "opacity": 0,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#262521",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#bcbbc1",
                  "flexDirection": "row",
                  "padding": 14,
                  "paddingBottom": 13.5,
                  "paddingLeft": 75,
                  "paddingTop": 13.5,
                  "text": Object {
                    "color": "#CECABA",
                    "fontSize": 12,
                  },
                }
              }
              testID="padView"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-start",
                      "flex": 1,
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                theme={
                  Object {
                    "colors": Object {
                      "black": "#242424",
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "grey": "rgba(0, 0, 0, 0.54)",
                          "primary": "#2196f3",
                          "searchBg": "#dcdce1",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "default": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "web": Object {
                          "error": "#ff190c",
                          "grey": "#393e42",
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                      "white": "#ffffff",
                    },
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#CECABA",
                      "fontSize": 12,
                    }
                  }
                  testID="listItemTitle"
                >
                  Exercise 1
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#262521",
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
                "paddingLeft": 75,
                "paddingTop": 40,
                "text": Object {
                  "color": "#CECABA",
                },
              }
            }
            testID="padView"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#CECABA",
                    "fontSize": 17,
                  }
                }
                testID="listItemTitle"
              >
                Cat_2
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingLeft": 16,
                }
              }
            />
            <View
              collapsable={false}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 24,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      󰅀
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Object {
              "maxHeight": "0%",
              "opacity": 0,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#262521",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#bcbbc1",
                  "flexDirection": "row",
                  "padding": 14,
                  "paddingBottom": 13.5,
                  "paddingLeft": 75,
                  "paddingTop": 13.5,
                  "text": Object {
                    "color": "#CECABA",
                    "fontSize": 12,
                  },
                }
              }
              testID="padView"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-start",
                      "flex": 1,
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                theme={
                  Object {
                    "colors": Object {
                      "black": "#242424",
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "grey": "rgba(0, 0, 0, 0.54)",
                          "primary": "#2196f3",
                          "searchBg": "#dcdce1",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "default": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "web": Object {
                          "error": "#ff190c",
                          "grey": "#393e42",
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                      "white": "#ffffff",
                    },
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#CECABA",
                      "fontSize": 12,
                    }
                  }
                  testID="listItemTitle"
                >
                  Exercise 2
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
